asv_bench/benchmarks/rolling.py

```diff
@@ -113,4 +113,35 @@ def time_quantile(self, constructor, window, dtype, percentile,
         self.roll.quantile(percentile, interpolation=interpolation)
 
 
+class PeakMemFixed(object):
+
+    params = (['max', 'min'])
+    param_names = ['method']
+
+    def setup(self, method):
+        N = 10**4
+        arr = 100 * np.random.random(N)
+        self.roll = pd.Series(arr).rolling(1000)
+
+    def peakmem_fixed(self, method):
+        for x in range(10000):
```

Comment:
Since this is a memory test, `sample_time` is ignored.  This test needs to be run ~10,000 times to see a 20% difference in peak memory usage.

Reply:
hmm this should show a difference even if run 1 time

running this too long is not great

why is this needed?

Reply:
Because of the base amount of memory being used, the test needs to be run a bunch of times for the memory leak to show as a 'significant' difference in ASV.  Even running it 10k times only shows a 20% difference in memory usage.

Reply:
you can lower the goal i think

it’s jot possible to run this 10k times

we have thousands of benchmarks

Reply:
Both tests together currently take about 45s.  If I try to reduce that to a few seconds (1000 loops) the memory use difference is <2%.

Maybe split the difference and have a ~22s for both tests with a ~10% memory difference?  I'm worried that anything less would not allow us to detect a memory leak in the future.

Reply:
reduce to just max or min then
add a comment in this tests goal

Reply:
22s is way too long 
these need to be pretty fast

Reply:
Having it just test one method would definitely help, but I'm not sure how useful of a test it would be if the loop count is drastically reduced and there is just a few percent memory difference.  No reasonable threshold would catch that and this test would serve no purpose.

Reply:
well as i said we have hundreds of benchmarks ; these only can detect non trivial memory regressions

so a few iterations ok

Reply:
Ok, I've reduced it to where the tests altogether take 2-3s to run, with ~1.7% memory difference.

Reply:
Does making the size of the series / window smaller allow for more iterations to showcase improvement without increasing runtime?

Reply:
I already made them smaller from the other `rolling` tests, and didn't see any big differences with a few parameters I tried, but I'll play around with it more and see what I can get.

Reply:
Ok, I've updated the ASV test with the best combination I could find that still ran in 2-3s.  This version removes the variable length testing (as that function is now out of scope for these changes) and a much smaller sized Series and window, allowing about 6000 loops to be run and to just barely hit the 10% threshold being used for regular continuous ASV testing.